ogc/query-with-cql-filters/src/main/java/com/esri/samples/query_with_cql_filters/QueryWithCqlFiltersController.java

```diff
@@ -0,0 +1,316 @@
+
+          // display result of query on the map
+          setInitialQueryOnOgcFeatureTable();
+
+        } else {
+          // show an alert dialog if there is a loading failure
+          new Alert(Alert.AlertType.ERROR, "Failed to load OGC Feature Collection Table: " +
+            ogcFeatureCollectionTable.getLoadError().getCause().getMessage()).show();
+        }
+      });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Gets the total extent of the OGC feature collection table, and uses a basic query parameters to return 3000 features
+   * from the service.
+   */
+  @FXML
+  private void setInitialQueryOnOgcFeatureTable() {
+
+    progressIndicator.setVisible(true);
+
+    // zoom to the dataset extent
+    if (datasetExtent != null && !datasetExtent.isEmpty()) {
+      mapView.setViewpointGeometryAsync(datasetExtent);
+    }
+
+    QueryParameters ogcFeatureExtentQueryParameters = new QueryParameters();
+    // create a query based on the current visible extent
+    ogcFeatureExtentQueryParameters.setGeometry(datasetExtent);
+    // set a limit of 3000 on the number of returned features per request, the default on some services could be as low as 10
+    ogcFeatureExtentQueryParameters.setMaxFeatures(3000);
+
+    try {
+      // populate and load the table with the query parameters
+      // set the clearCache parameter to true to clear existing table entries, set the outfields parameter to null to request all fields
+      ogcFeatureCollectionTable.populateFromServiceAsync(ogcFeatureExtentQueryParameters, true, null).addDoneListener(() -> {
+        progressIndicator.setVisible(false);
+        applyQueryButton.setDisable(false);
+        // display number of features returned
+        setFeatureNumberLabelText();
+      });
+
+      // handle UI
+      revertToInitialQueryButton.setDisable(true);
+      textField.setText("3000");
+      // reset combo box and show original prompt text
+      comboBox.getSelectionModel().clearSelection();
+      comboBox.setButtonCell(new ListCell<>() {
+        @Override
+        protected void updateItem(String item, boolean empty) {
+          setText(item);
+        }
+      });
+
+    } catch (Exception exception) {
+      exception.printStackTrace();
+      new Alert(Alert.AlertType.ERROR, exception.getMessage()).show();
+    }
+  }
+
+  /**
+   * Populates features from provided query parameters, and displays the result on the map.
+   */
+  @FXML
+  private void query() {
+
+    QueryParameters cqlQueryParameters = new QueryParameters();
+    // set the query parameter's where clause with the CQL query in the combo box
+    cqlQueryParameters.setWhereClause(comboBox.getSelectionModel().getSelectedItem());
+    // set the max features to the number entered in the text field
+    cqlQueryParameters.setMaxFeatures(Integer.parseInt(textField.getText()));
```

Comment:
I think you could get a number format exception here if the text field can't be converted into an integer.

Reply:
I've added a try catch here to catch this: thanks!